src/Api/EncodedStringTest.cpp

```diff
@@ -0,0 +1,521 @@
+  std::string decoded = DecodeString(0x0000000044332211, 0, 0, 0, 0, 0, 0, 0, nullptr, 0);
+
+  std::string expected{0x11, 0x22, 0x33, 0x44};
+
+  EXPECT_EQ(expected, decoded);
+}
+
+TEST(EncodedString, DecodeNineByteString) {
+  std::string decoded =
+      DecodeString(0x8877665544332211, 0x0000000000000099, 0, 0, 0, 0, 0, 0, nullptr, 0);
+
+  std::string expected{
+      0x11, 0x22, 0x33, 0x44, 0x55, 0x66, 0x77, static_cast<char>(0x88), static_cast<char>(0x99)};
+
+  EXPECT_EQ(expected, decoded);
+}
+
+TEST(EncodedString, Decode64ByteString) {
+  std::string decoded = DecodeString(0x8877665544332211, 0x8877665544332211, 0x8877665544332211,
+                                     0x8877665544332211, 0x8877665544332211, 0x8877665544332211,
+                                     0x8877665544332211, 0x8877665544332211, nullptr, 0);
+  std::string expected{
+      0x11, 0x22, 0x33, 0x44, 0x55, 0x66, 0x77, static_cast<char>(0x88),  // 1
+      0x11, 0x22, 0x33, 0x44, 0x55, 0x66, 0x77, static_cast<char>(0x88),  // 2
+      0x11, 0x22, 0x33, 0x44, 0x55, 0x66, 0x77, static_cast<char>(0x88),  // 3
+      0x11, 0x22, 0x33, 0x44, 0x55, 0x66, 0x77, static_cast<char>(0x88),  // 4
+      0x11, 0x22, 0x33, 0x44, 0x55, 0x66, 0x77, static_cast<char>(0x88),  // 5
+      0x11, 0x22, 0x33, 0x44, 0x55, 0x66, 0x77, static_cast<char>(0x88),  // 6
+      0x11, 0x22, 0x33, 0x44, 0x55, 0x66, 0x77, static_cast<char>(0x88),  // 7
+      0x11, 0x22, 0x33, 0x44, 0x55, 0x66, 0x77, static_cast<char>(0x88),  // 8
+  };
+
+  EXPECT_EQ(expected, decoded);
+}
+
+TEST(EncodedString, Decode65ByteStringIncludingOneAdditionalCharacter) {
+  std::vector<uint64_t> additional_char;
+  additional_char.push_back(0x0000000000000011);
+  std::string decoded =
+      DecodeString(0x8877665544332211, 0x8877665544332211, 0x8877665544332211, 0x8877665544332211,
+                   0x8877665544332211, 0x8877665544332211, 0x8877665544332211, 0x8877665544332211,
+                   additional_char.data(), additional_char.size());
+  std::string expected{0x11, 0x22, 0x33, 0x44, 0x55, 0x66, 0x77, static_cast<char>(0x88),  // 1
+                       0x11, 0x22, 0x33, 0x44, 0x55, 0x66, 0x77, static_cast<char>(0x88),  // 2
+                       0x11, 0x22, 0x33, 0x44, 0x55, 0x66, 0x77, static_cast<char>(0x88),  // 3
+                       0x11, 0x22, 0x33, 0x44, 0x55, 0x66, 0x77, static_cast<char>(0x88),  // 4
+                       0x11, 0x22, 0x33, 0x44, 0x55, 0x66, 0x77, static_cast<char>(0x88),  // 5
+                       0x11, 0x22, 0x33, 0x44, 0x55, 0x66, 0x77, static_cast<char>(0x88),  // 6
+                       0x11, 0x22, 0x33, 0x44, 0x55, 0x66, 0x77, static_cast<char>(0x88),  // 7
+                       0x11, 0x22, 0x33, 0x44, 0x55, 0x66, 0x77, static_cast<char>(0x88),  // 8
+                       0x11};
+
+  EXPECT_EQ(expected, decoded);
+}
+
+TEST(EncodedString, Decode80ByteStringIncluding16AdditionalCharacters) {
+  std::vector<uint64_t> additional_char;
```

Comment:
Rename `additional_char` (e.g., to `additional_chunks`).

Reply:
Done.